### PR TITLE
support "~timestampu" and  "~timestampz"

### PR DESCRIPTION
**Pull request by shandilya3**
~timestampu - the current device time in epoch format (seconds since epoch).
~timestampz - the current device time in ISO 8601 data elements and interchange format.
~sdkver - the current Adobe Experience Platform SDKs version string.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [X] I have made any necessary test changes and all tests pass.
- [X] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/DecisioningEngine/createContextProvider.js [hunk 1]
```diff
@@ -74,7 +77,8 @@ export default ({ eventRegistry, window }) => {
     return {
       ...coreGlobalContext,
       ...getTimeContext(),
-      window: getWindowContext()
+      window: getWindowContext(),
+      "~sdkver": version
```
Comment (carterworks) [inline src/components/DecisioningEngine/createContextProvider.js:81]: The version can be imported from `src/constants/libraryVersion.js`. No need to add a dependency and plugin to the build.

**Review by carterworks — CHANGES_REQUESTED** [review]

File: src/components/DecisioningEngine/createContextProvider.js (hunk 1 above)
Comment (shandilya3) [reply src/components/DecisioningEngine/createContextProvider.js:81]: Thanks @carterworks  for bringing this up. I was also looking for existing use, guess didn't search with matching keyword.

File: test/unit/specs/components/DecisioningEngine/decisioningContext.sdkVersion.spec.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      })
+    );
+  });
+
+  it("does not satisfy rule due to unmatched dk version", () => {
+    setupResponseHandler(applyResponse, mockWindow({}), {
+      definition: {
+        key: "~sdkver",
+        matcher: "eq",
+        values: ["2.18.0-beta.0"]
```
Comment (jasonwaters) [inline test/unit/specs/components/DecisioningEngine/decisioningContext.sdkVersion.spec.js:48]: Guessing this test will fail every time the SDK version is updated.  I suggest mocking it instead.

File: src/components/DecisioningEngine/createContextProvider.js (hunk 1 above)
Comment (jasonwaters) [reply src/components/DecisioningEngine/createContextProvider.js:81]: +1

File: test/unit/specs/components/DecisioningEngine/decisioningContext.sdkVersion.spec.js (hunk 2 above)
Comment (shandilya3) [reply test/unit/specs/components/DecisioningEngine/decisioningContext.sdkVersion.spec.js:48]: Testing above with real libraryVersion, that will always match.
 This one is false test case.

**Review by carterworks — APPROVED** [review]

File: test/unit/specs/components/DecisioningEngine/decisioningContext.sdkVersion.spec.js (hunk 2 above)
Comment (jasonwaters) [reply test/unit/specs/components/DecisioningEngine/decisioningContext.sdkVersion.spec.js:48]: 😆 ah that what i get for not reading it closely enough.

**Review by jasonwaters — APPROVED** [review]

**Pull request merged**